core/src/com/ericc/the/game/systems/logic/FieldOfViewSystem.java

```diff
@@ -81,44 +87,69 @@ public void update(float deltaTime) {
         }
     }
 
+    /**
+     * A decorator-like function to call with less arguments.
+     * @param pos a Position Component of an Entity
+     * @param fov a Field of View Component of the same Entity
+     */
     private void processEntity(PositionComponent pos, FieldOfViewComponent fov) {
-        updateFOV(visibleMapArea.top, visibleMapArea.bottom, visibleMapArea.left, visibleMapArea.right, pos.x, pos.y, fov);
+        updateFOV(pos.x, pos.y, fov);
     }
 
-    private void clearFOV(int top, int bottom, int left, int right, FieldOfViewComponent fov) {
-        for (int y = top; y >= bottom; --y) {
-            for (int x = left; x <= right; ++x) {
+    /**
+     * Clears the given fov of any Entity but restricts clearing
+     * only to visible map area (by player)
+     * to improve performance.
+     * @param fov a Field of View Component of any Entity
+     */
+    private void clearFOV(FieldOfViewComponent fov) {
```

Comment:
teraz jak dodałeś komentarz to się zacząłem zastanawiać czy to jest na pewno dobrze, bo chyba chcemy żeby AI działało sobie nawet jak jest poza widoczną częścią mapy (bo np. jest potwór który widzi całą mapę albo jego wzrok przenika przez ściany). Ale to najwyżej do tego wrócimy kiedy indziej

Reply:
Dodaje issue